### ISSUE TITLE
Request help on debuging the running speed

### ISSUE THREAD

**Issue by Remember2018** [tensorpack/tensorpack#1132]
I'm using the `FasterRCNN` module in Tensorpack and I modified the network by adding some other branch. The training data was saved in a NFS file system (in `png` format and `xml` format) and I run the code in a node of a cluster. The node has 4 K80 GPUs. When I run the `FasterRCNN` in this system. It seems the executation is slow, please see the following log:

        11%|#         |53/500[03:00<25:20, 0.29it/s]
        12%|#2        |61/500[03:20<24:53, 0.29it/s]
        25%|##5       |125/500[06:03<18:34, 0.34it/s]
        26%|##6       |130/500[06:20<18:19, 0.34it/s]
        40%|###9      |198/500[09:04<13:43, 0.37it/s]
        41%|####1     |205/500[09:20<13:24, 0.37it/s]
        54%|#####4    |271/500[12:04<09:54, 0.38it/s]
        56%|#####5    |278/500[12:20<09:36, 0.38it/s] 
       70%|#######   |352/500[15:20<06:09, 0.40it/s]
       85%|########4 |424/500[18:05<03:03, 0.41it/s]
       86%|########6 |430/500[18:20<02:48, 0.41it/s]
       100%|##########|500/500[21:29<00:00, 0.39it/s][32m[0407 08:33:58 @base.py:285][0m Epoch 1 (global_step 500) finished, time:21 minutes 29 seconds.

But when I check the GPU executation, it shows `100%` execution at most of time. But sometimes  three of 4 GPUs has `0%` execution load and only one shows `100%` load.  
I'm not sure how to speed up the network training, it seems the execution of the network is slow compared to the original `FasterRCNN` in Tensorpack. I'm not sure if this related to the data prefetching because of the busy NFS file system in the cluster. 

And I've tried to save all the images and annotations in the memory to speed up the data loading, but it failed due to the memory problem. Could you please give some advice on how to speed up the network training in this case? Thanks.

**Comment by ppwwyyxx** [comment]
Have you seen: https://github.com/tensorpack/tensorpack/blob/master/examples/FasterRCNN/NOTES.md ?

> If CuDNN warmup is on, the training will start very slowly, until about 10k steps (or more if scale augmentation is used) to reach a maximum speed. As a result, the ETA is also inaccurate at the beginning. CuDNN warmup is by default on when no scale augmentation is used.

You can set TF_AUTOTUNE_THRESHOLD to 0 in `config.py` to disable autotune.

**Comment by Remember2018** [comment]
Hi @ppwwyyxx , thanks a lot for your kind suggestions!
I've tried to modify the `os.environ['TF_CUDNN_USE_AUTOTUNE'] = '0'` and `os.environ['TF_AUTOTUNE_THRESHOLD'] = '0'` in `config.py`, and it has some speed up and the GPU memory usage seems decreased about ~2000M(from ~10000M to ~8000M). And the data loading seems costs `300%` CPU usage compared to the previous version.  Currently the log in the cluster is as follows:

     15%|#5        |77/500[03:02<16:39, 0.42it/s]
     17%|#7        |86/500[03:20<16:18, 0.42it/s]
     35%|###4      |174/500[06:02<11:28, 0.47it/s]
     37%|###6      |183/500[06:20<11:09, 0.47it/s]
     54%|#####4    |271/500[09:03<07:35, 0.50it/s]
     56%|#####6    |280/500[09:20<07:17, 0.50it/s]
     74%|#######3  |369/500[12:04<04:11, 0.52it/s]
     75%|#######5  |377/500[12:20<03:55, 0.52it/s]
     93%|#########3|467/500[15:05<01:02, 0.53it/s]
     95%|#########4|474/500[15:20<00:49, 0.53it/s]
    100%|##########|500/500[16:37<00:00, 0.50it/s][32m[0407 09:22:41 @base.py:285][0m Epoch 1 (global_step 500) finished, time:16 minutes 37 seconds.

And I've also tried the same code in local machine with 2 1080Ti GPUs, and also set the flags of CUDNN is `0`. The speed has ~`1.50it/s`. So maybe the bottleneck is in the data loading?

The dataset has about `50000` images and the size of each image is `800x800x3`. I saved the images into a python dictionary which has size of about 100G. I loaded the dictionary in the `get_train_dataflow()` and replace the image reading according to the original `roidb` information with the memory reading from the dictionary. The `free` command show the node has ~120G available memory, but when loading the dictionary into the memory, it said memory loading error. Does Tensorpack support the data prefetching from the memory?
